calculator.js

```diff
@@ -0,0 +1,96 @@
+(function() {
+    "use strict";
+
+    var result = 0;
+    var callScreen = require('calculator.output.Screen');
+    var screen = new callScreen(document.getElementById('screen'));
+
+    var buttons = document.querySelector('.allButtons');
+    var numberButtons = buttons.querySelectorAll('.number');
+
+    for (var i = 0; i <= numberButtons.length - 1; i++) {
+
+        numberButtons[i].onclick = function() {
+            var buttonDigit = parseInt(this.innerHTML);
+            screen.addDigit(buttonDigit);
+        }
+    };
+
+    var operationButtons = buttons.querySelectorAll('.operation');
+
+    for (var i = 0; i < operationButtons.length; i++) {
+        var isInitialAction = true;
+        var prevOperation;
+
+        operationButtons[i].onclick = function() {
+            var currNumber = screen.getNumber();
+
+
+            if (isInitialAction) {
+                result = currNumber;
+                isInitialAction = false;
+
+            } else {
+
+                switch (prevOperation) {
+
+                    case '+':
+                        result = Number(result) + Number(currNumber);
+                        break;
+
+                    case '-':
+                        result = Number(result) - Number(currNumber);
+                        break;
+
+                    case 'x':
+                        result = Number(result) * Number(currNumber);
+                        break;
+
+                    case '/':
+                        result = Number(result) / Number(currNumber);
+                        break;
+                     case '=': 
+                        result = currNumber;
+                        break;
+                }
+            }
+            screen.resetOnNextInput();
+            screen.setNumber(result);
+            prevOperation = this.innerHTML;
+        }
+    }
+
+    var newNum = require('memory.num');
```

Comment:
Do not use such illogical names -- newNum does not mean anything. Use normal class names, for example you should use **Memory** here.